src/sources/kubernetes_logs/mod.rs

```diff
@@ -123,50 +164,69 @@ pub struct Config {
 
     /// Ignore files with a data modification date older than the specified number of seconds.
     #[serde(default)]
+    #[configurable(metadata(docs::type_unit = "seconds"))]
+    #[configurable(metadata(docs::examples = 600))]
     ignore_older_secs: Option<u64>,
 
     /// Max amount of bytes to read from a single file before switching over
     /// to the next file.
+    ///
     /// This allows distributing the reads more or less evenly across
     /// the files.
+    #[configurable(metadata(docs::type_unit = "bytes"))]
     max_read_bytes: usize,
 
-    /// The maximum number of bytes a line can contain before being discarded. This protects
-    /// against malformed lines or tailing incorrect files.
+    /// The maximum number of bytes a line can contain before being discarded.
+    ///
+    /// This protects against malformed lines or tailing incorrect files.
+    #[configurable(metadata(docs::type_unit = "bytes"))]
     max_line_bytes: usize,
 
-    /// How many first lines in a file are used for fingerprinting.
+    /// The number of lines to read for generating the checksum.
+    ///
+    /// If your files share a common header that is not always a fixed size,
+    ///
+    /// If the file has less than this amount of lines, it won’t be read at all.
+    #[configurable(metadata(docs::type_unit = "lines"))]
     fingerprint_lines: usize,
 
     /// This value specifies not exactly the globbing, but interval
     /// between the polling the files to watch from the `paths_provider`.
+    ///
     /// This is quite efficient, yet might still create some load of the
     /// file system; in addition, it is currently coupled with checksum dumping
     /// in the underlying file server, so setting it too low may introduce
     /// a significant overhead.
-    glob_minimum_cooldown_ms: usize,
+    #[serde_as(as = "serde_with::DurationMilliSeconds<u64>")]
+    glob_minimum_cooldown_ms: Duration,
```

Comment:
Just a question, where is this field getting it's default value of 60,000 from? The `default_glob_minimum_cooldown_ms` function isn't mentioned here.. does `serde` magically pick it up based on the name?

Reply:
That's fine for creating a default value, but it won't work if  deserializing the config? If that field is missing won't serde just error?

Reply:
Ah, no ... `#[serde(default)]`..